cmd/thanos/compact.go

```diff
@@ -90,63 +90,82 @@ func registerCompact(app *extkingpin.App) {
 	})
 }
 
-func runCompact(
-	g *run.Group,
-	logger log.Logger,
-	tracer opentracing.Tracer,
-	reg *prometheus.Registry,
-	component component.Component,
-	conf compactConfig,
-	flagsMap map[string]string,
-) error {
-	deleteDelay := time.Duration(conf.deleteDelay)
-	halted := promauto.With(reg).NewGauge(prometheus.GaugeOpts{
+type CompactMetrics struct {
+	halted                      prometheus.Gauge
+	retried                     prometheus.Counter
+	iterations                  prometheus.Counter
+	cleanups                    prometheus.Counter
+	partialUploadDeleteAttempts prometheus.Counter
+	blocksCleaned               prometheus.Counter
+	blockCleanupFailures        prometheus.Counter
+	blocksMarked                *prometheus.CounterVec
+	garbageCollectedBlocks      prometheus.Counter
+}
+
+func newCompactMetrics(reg *prometheus.Registry, deleteDelay time.Duration) *CompactMetrics {
+	_ = promauto.With(reg).NewGaugeFunc(prometheus.GaugeOpts{
+		Name: "thanos_delete_delay_seconds",
+		Help: "Configured delete delay in seconds.",
+	}, func() float64 {
+		return deleteDelay.Seconds()
+	})
+
+	m := new(CompactMetrics)
```

Comment:
Let's stick to `m := CompactMetrics{}` for consistency (we never use new). I think we don't need pointer too, that's why new is not needed